client/src/router.tsx

```diff
@@ -174,6 +181,9 @@ const routes = createRoutesFromElements(
             }
           />
         </Route>
+        <Route path="search" element={<SearchEmptyRoute.RouteComponent />}>
+          <Route path=":query" element={<SearchEmptyRoute.RouteComponent />} />
```

Comment:
Since you're not rendering an `<Outlet />` on the main search page, could you just pass `null` here?

```suggestion
          <Route path=":query" element={null} />
```

Reply:
Ended up doing a refactor after removing the key on the top-level `<Outlet />` which is a transparent change requiring no other fixes (playlist switching looks the same before/after).

That refactoring is in https://github.com/apollographql/spotify-showcase/pull/87/commits/d2bd8bc6bdcd9d0ad6dd717e35d038bee1967a59: I could remove the input focusing in `useEffect` and use an `<Outlet />` for the search results 🎉